src/github.ts

```diff
@@ -0,0 +1,81 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { RestEndpointMethodTypes } from "@octokit/plugin-rest-endpoint-methods";
+
+type ClientType = ReturnType<typeof github.getOctokit>;
+
+export function getClient(token: string): ClientType {
+  return github.getOctokit(token);
+}
+
+export type Output = {
+  title: string;
+  summary: string;
+  annotations: Annotation[];
+};
+
+export type Annotation = {
+  path: string;
+  start_line: number;
+  end_line: number;
+  annotation_level: AnnotationLevel;
+  message: string;
+  title: string;
+  raw_details: string;
+};
+
+export type AnnotationLevel = "notice" | "warning" | "failure";
+
+export async function createCheck(
+  client: ClientType,
+  name: string,
+  annotations: Annotation[]
+): Promise<void> {
+  const output = buildOutput(annotations);
+  const pullRequest = github.context.payload.pull_request;
+  const head_sha = (pullRequest && pullRequest.head.sha) || github.context.sha;
+  const status = "completed";
+  const failures = output.annotations.filter((a) => {
+    return a.annotation_level === "failure";
+  });
+  const conclusion = failures.length > 0 ? "failure" : "success";
+
+  await client.rest.checks.create({
+    ...github.context.repo,
+    name,
+    head_sha,
+    status,
+    conclusion,
+    output,
+  });
+}
+
+function buildOutput(annotations: Annotation[]): Output {
+  const annotationsCount = annotations !== null ? annotations.length : 0;
+  const title = `${annotationsCount} result(s) found by grep`;
+  const summary = "";
+
+  if (annotationsCount > 50) {
```

Comment:
Can that be put into a variable like `MAX_ANNOTATIONS`?

Reply:
Also, is this limit imposed by Github?

Reply:
Sure on the constant.

Their API only accepts 50 at a time. If you have more you are supposed to create the check with 50 then perform updates to add the rest (with fewer than 50 at a time).

It's not terrible code to write, I just left it as a TODO to implement later.